lib/src/main/java/com/auth0/android/lock/events/DatabaseSignUpEvent.java

```diff
@@ -55,7 +52,7 @@ public String getPassword() {
         return password;
     }
 
-    public void setExtraFields(@NonNull Map<String, String> customFields) {
+    public void setExtraFields(@NonNull Map<String, Object> customFields) {
```

Comment:
The only "breaking change" is a change in a signature, but goes from specific (String) to generic (Object) so, there shouldn't be any breaking change here. @jimmyjames please bless this :D .

Reply:
It would be a breaking change, since `Map<String, String>` is actually not a subclass of `Map<String, Object>`. So given the following code:

```java
Map<String, String> m1 = new HashMap<>();
m1.put("key", "val");
dse.setExtraFields(m1);
```

Would result in a compilation failure like
```
error: incompatible types: Map<String,String> cannot be converted to Map<String,Object>
```

[This article](https://nofluffjuststuff.com/magazine/2016/09/time_to_really_learn_generics_a_java_8_perspective#object_collection) contains more information about Collections and type invariance.

Reply:
To make this backwards-compatible, I think you could use a bounded wildcard of `? extends Object`:

```java
public void setExtraFields(@NonNull Map<String, ? extends Object> customFields) {
```

Reply:
I'll add a separate method 👍